File: app/src/main/java/com/example/mode/mode_proxypattern/MainActivity.java
package com.example.mode.mode_proxypattern;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.kotlin.R;

/**
 * 代理模式
 * 给目标对象提供一个代理对象，并由代理对象控制对目标对象的引用
 */
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rxjava1);
        Proxy proxy = new Proxy();
        proxy.buyMac();
    }
}

File: app/src/main/java/com/example/kotlin/demo/MainActivity.kt
package com.example.kotlin.demo

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.kotlin.R
import kotlinx.android.synthetic.main.activity_rxjava1.*

class MainActivity:AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rxjava1)
        tv_init.setOnClickListener{
            //val intent = Intent(this@MainActivity.c)
        }
        val user:User = User()

        user?.name
        user?.age
        user?.let{
            user.name
            user.age
        }


    }

}